cmd/esc/cli/env_clone.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023, Pulumi Corporation.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pulumi/esc/cmd/esc/cli/client"
+	"github.com/spf13/cobra"
+)
+
+func newEnvCloneCmd(env *envCommand) *cobra.Command {
+	var (
+		preserveHistory         bool
+		preserveAccess          bool
+		preserveEnvironmentTags bool
+		preserveRevisionTags    bool
+	)
+
+	cmd := &cobra.Command{
+		Use:   "clone [<org-name>/]<project-name>/<environment-name> [<project-name>/]<environment-name>",
+		Args:  cobra.MaximumNArgs(2),
+		Short: "Clone an existing environment into a new environment.",
+		Long: "Clone an existing environment into a new environment.\n" +
+			"\n" +
+			"This command clones an existing environment with the given identifier into a new environment.\n" +
+			"If a project is omitted from the new environment identifier the new environment will be created\n" +
+			"within the same project as the environment being cloned.\n",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			if err := env.esc.getCachedClient(ctx); err != nil {
+				return err
+			}
+
+			ref, args, err := env.getNewEnvRef(ctx, args)
+			if err != nil {
+				return err
+			}
+			if ref.version != "" {
+				return fmt.Errorf("the clone command does not accept versions")
+			}
+
+			var destProject string
+			destName := args[0]
+			destParts := strings.Split(args[0], "/")
```

Comment:
I think it may be easier to use `Cut` here, and then check if the second part exists - if so, it's project/env otherwise just env.
Split works also, but then you'll need to return an error for len > 2

a part of me is also wondering if we could reuse the existing ref parsing methods but since we're not parsing org it'll be slightly different 🤷 probably fine to not reuse here

Reply:
yeah I was thinking of reusing that function but a lot of that logic has things around optional org and project where as this will never have an org prefix so I wanted to remove the ability for that to be confused.